src/services/leafletLayerHelpers.js

```diff
@@ -213,7 +223,19 @@ angular.module("leaflet-directive")
 
                 params.options.url = params.url;
 
-                return L.esri.featureLayer(params.options);
+                var layer = L.esri.featureLayer(params.options);
+                var load = function() {
+                    if(isDefined(params.options.loadedDefer)) {
+                        params.options.loadedDefer.resolve();
+                    }
+                };
+                layer.on('loading', function() {
+                    params.options.loadedDefer = $q.defer();
```

Comment:
Is this not dangerous to allow anyone to have access to loadedDefer to resolve it when they want to?